messaging/kafka-avro-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/BaseKafkaAvroGroupIdIT.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.ts.messaging.kafka;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.restassured.http.ContentType;
+
+abstract class BaseKafkaAvroGroupIdIT {
+    private static final int TIMEOUT_SEC = 25;
+    private static final int EVENTS_AMOUNT = 5;
+
+    private String endpoint;
+    private Client client = ClientBuilder.newClient();
+    private List<String> receive = new CopyOnWriteArrayList<>();
+    private boolean completed;
+    private Random rand = new Random();
+
+    @Test
+    public void testAlertMonitorEventStream() throws InterruptedException {
+        GivenSomeStockPrices(getAppA(), EVENTS_AMOUNT);
+        AndApplicationEndpoint(getEndpoint(getAppA()) + "/stock-price/stream");
+        whenRequestSomeEvents(EVENTS_AMOUNT);
+        thenVerifyAllEventsArrived();
+        // Application B should have a different auto-generated group ID so,
+        // double the number of events
```

Comment:
I could not find a clear way to do it. So I have opened a Zulip conversation
Link: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Kafka/near/254735554

Reply:
On the other hand, retrieve and check that the groupID property is different between the appA and B is fine, but this will only prove that the property is well settled. I prefer to do the end-2-end verification.